module/VuFind/src/VuFind/ChannelProvider/ChannelLoader.php

```diff
@@ -69,6 +70,28 @@ public function __construct(
     ) {
     }
 
+    /**
+     * Add configuration values needed by the templates to the view context
+     *
+     * @param array $context String-keyed map of values for the View
+     *
+     * @return array
+     */
+    protected function addConfigToContext($context)
+    {
+        for ($i = 0; $i < count($context['channels']); $i++) {
+            [, $configSection] = explode(':', $context['channels'][$i]['providerId'] . ':');
+            $config = $this->config->$configSection;
```

Comment:
Maybe this should have a default value:


```suggestion
            $config = $this->config->$configSection ?? [];
```

...which also leads me to wonder if it's worth taking the time to revise this code to use arrays instead of Config objects, since we're trying to move away from the objects. Probably better done in a separate PR, but maybe something I could help with, time permitting.

Reply:
My [earlier comment about providing a default value](https://github.com/vufind-org/vufind/pull/4557#discussion_r2428965755) still applies here. Let's not get bogged down in factoring out Config objects unless it's easy -- but that is something we should consider doing as a follow-up action.

Reply:
There's also something more broadly incorrect here -- this `$config` variable is never used, and it is followed by references to an `$options` variable that is never defined.

Reply:
Some of this logic (e.g. the minimum batch size loop) also seems to duplicate behavior in the BatchTrait; is that expected/intended?